app/components/Authentication/AwsEc2/AwsEc2.jsx

```diff
@@ -1,27 +1,450 @@
+import Subheader from 'material-ui/Subheader';
+import ActionAccountBox from 'material-ui/svg-icons/action/account-box';
+import ActionDelete from 'material-ui/svg-icons/action/delete';
+import ActionDeleteForever from 'material-ui/svg-icons/action/delete-forever';
+// Styles
+import styles from './awsec2.css';
+import sharedStyles from '../../shared/styles.css';
+import { green500, green400, red500, red300, yellow500, white } from 'material-ui/styles/colors.js';
+import Checkbox from 'material-ui/Checkbox';
+import { callVaultApi, tokenHasCapabilities } from '../../shared/VaultUtils.jsx';
+// Misc
+import _ from 'lodash';
+import update from 'immutability-helper';
+import Avatar from 'material-ui/Avatar';
+import { browserHistory } from 'react-router'
+import PolicyPicker from '../../shared/PolicyPicker/PolicyPicker.jsx'
+import VaultObjectDeleter from '../../shared/DeleteObject/DeleteObject.jsx'
+
+function snackBarMessage(message) {
+    let ev = new CustomEvent("snackbar", { detail: { message: message } });
+    document.dispatchEvent(ev);
+}
 
 export default class AwsEc2AuthBackend extends React.Component {
+
+    ec2ConfigSchema = {
+        access_key: '',
+        endpoint: undefined,
+        secret_key: ''
+    };
+
+    roleConfigSchema = {
+        role: '',
+        bound_ami_id: undefined,
+        bound_account_id: undefined,
+        bound_region: undefined,
+        bound_vpc_id: undefined,
+        bound_subnet_id: undefined,
+        bound_iam_role_arn: undefined,
+        bound_iam_instance_profile_arn: undefined,
+        role_tag: undefined,
+        ttl: undefined,
+        period: undefined,
+        policies: [],
+        allow_instance_migration: undefined,
+        disallow_reauthentication: undefined
+    }
+
     constructor(props) {
         super(props);
         this.state = {
-
+            baseUrl: `/auth/aws-ec2/${this.props.params.namespace}/`,
+            baseVaultPath: `auth/${this.props.params.namespace}`,
+            ec2Roles: [],
+            configObj: this.ec2ConfigSchema,
+            newConfigObj: this.ec2ConfigSchema,
+            newRoleConfig: this.roleConfigSchema,
+            newSecretBtnDisabled: false,
+            openNewRoleDialog: false,
+            openEditRoleDialog: false,
+            deleteUserPath: ''
         };
 
+        _.bindAll(
+            this,
+            'listEc2Roles',
+            'getEc2AuthConfig',
+            'createUpdateConfig',
+            'createUpdateRole'
+        );
+
+    }
+
+    listEc2Roles() {
+        tokenHasCapabilities(['list'], `${this.state.baseVaultPath}/role`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/role`, { list: true }, null)
+                    .then((resp) => {
+                        let roles = resp.data.data.keys;
+                        this.setState({ ec2Roles: _.valuesIn(roles) });
+                    })
+                    .catch(snackBarMessage);
```

Comment:
Vault will return 404 here if no roles are configured. I recommend handling the exception to suppress the snackbar message in that case